src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -348,7 +348,7 @@ public void AddInput(string command)
                 }
 
                 FinishCurrentSubmissionInput();
-                _window._history.Add(_window._currentLanguageBuffer.CurrentSnapshot.GetExtent());
+                _window._history.Add(_window._currentLanguageBuffer.CurrentSnapshot.GetExtent().TrimEnd());
```

Comment:
I'm not sure this is the right fix actually. It should suffice to move this line before the call to FinishCurrentSubmissionInput, which appends the final ```\r\n```.